Plugin~/Src/MeshSync/SceneGraph/msScene.cpp

```diff
@@ -231,27 +233,62 @@ void Scene::sanitizeObjectName(std::string& name)
     }
 }
 
-void Scene::import(const SceneImportSettings& cv)
-{
+std::vector<EntityConverterPtr> Scene::getConverters(const SceneImportSettings& cv, const SceneSettings& settings, bool invert) {
     // receive and convert assets
     bool flip_x = settings.handedness == Handedness::Right || settings.handedness == Handedness::RightZUp;
     bool swap_yz = settings.handedness == Handedness::LeftZUp || settings.handedness == Handedness::RightZUp;
 
     std::vector<EntityConverterPtr> converters;
     if (settings.scale_factor != 1.0f) {
-        float scale = 1.0f / settings.scale_factor;
+        float scale;
+
+        if (invert) {
+            scale = settings.scale_factor;
+        }
+        else {
+            scale = 1.0f / settings.scale_factor;
+        }
+
         converters.push_back(ScaleConverter::create(scale));
     }
     if (flip_x) {
         converters.push_back(FlipX_HandednessCorrector::create());
     }
     if (swap_yz) {
-        if (cv.zup_correction_mode == ZUpCorrectionMode::FlipYZ)
-            converters.push_back(FlipYZ_ZUpCorrector::create());
-        else if (cv.zup_correction_mode == ZUpCorrectionMode::RotateX)
-            converters.push_back(RotateX_ZUpCorrector::create());
+        if (cv.zup_correction_mode == ZUpCorrectionMode::FlipYZ) {
+            if (invert) {
+                // TODO: Could write a new FlipYZ_ZUpCorrector that flips back, for now just rotate 3 more times to get back to original:
+                for (int i = 0; i < 3; i++)
+                {
+                    converters.push_back(FlipYZ_ZUpCorrector::create());
+                }
+            }
+            else {
+                converters.push_back(FlipYZ_ZUpCorrector::create());
+            }
+        }
+        else if (cv.zup_correction_mode == ZUpCorrectionMode::RotateX) {
+            if (invert) {
+                // TODO: Could write a new RotateX_ZUpCorrector that rotates back, for now just rotate 3 more times to get back to original:
+                for (int i = 0; i < 3; i++)
```

Comment:
I don't understand why doing the same rotation three times would be the same as doing the inverse rotation. Is this the case ?

Reply:
Yeah, the converter has a hardcoded rotation of 90 degrees, i just rotate 3 more times at 90 degrees to get back where it was. After this is merged I'm going to refactor that to set the rotation in its constructor or something.

Reply:
It does some swapping and flipping while doing the rotation, and the transformation isn't the same for position, rotation and scale.
Are you sure that doing this 3 times will invert all of them to where they were ?
```
    if (e.isRoot()) {
        e.position = flip_z(swap_yz(e.position));
        e.rotation = flip_z(swap_yz(e.rotation)) * mu::rotate_x(-90.0f * mu::DegToRad);
        e.scale = swap_yz(e.scale);
    }
```

Might be worthy to turn this into a unit test.

Reply:
I've added unit tests to make sure this works. Turns out the RotateX_ZUpCorrector needs 8 transformations because of the z flipping. I hadn't used that much because it's buggy anyway.